packages/legacy-scripting-runner/middleware-factory.js

```diff
@@ -203,18 +203,21 @@ const createBeforeRequest = app => {
 
 const createAfterResponse = app => {
   const authType = _.get(app, 'authentication.type');
+  const autoRefresh =
+    _.get(app, 'legacy.authentication.oauth2Config.autoRefresh') ||
+    _.get(app, 'authentication.oauth2Config.autoRefresh');
 
-  const sessionAuthCheckResponse = (response, z) => {
+  const throwForStaleAuth = (response, z) => {
     if (response.status === 401) {
-      throw new z.errors.RefreshAuthError('Session key needs refreshing');
+      throw new z.errors.RefreshAuthError('Authentication needs refreshing');
     }
     return response;
   };
 
   let afterResponse;
 
-  if (authType === 'session') {
-    afterResponse = sessionAuthCheckResponse;
+  if (authType === 'session' || (authType === 'oauth2' && autoRefresh)) {
+    afterResponse = throwForStaleAuth;
```

Comment:
For some reason the legacy runner relied on `throwForStaleAuth` for oauth2 only, but now it has to do so for both session and oauth2.